library/oneview_id_pools_ipv4_range.py

```diff
@@ -91,43 +91,72 @@ def __init__(self):
 
         super(IdPoolsIpv4RangeModule, self).__init__(additional_arg_spec=additional_arg_spec,
                                                      validate_etag_support=True)
-
+        self.connection_templates = self.oneview_client.connection_templates
         self.resource_client = self.oneview_client.id_pools_ipv4_ranges
 
     def execute_module(self):
         resource = None
+        # If Range URI is provided then it sets the resource client
         if self.data.get('uri'):
-            resource = self.resource_client.get(self.data.get('uri'))
+            resource = self.resource_client.get_by_uri(self.data.get('uri'))
+        # Do preliminary check before creating a new range
         elif self.data.get('subnetUri') and self.data.get('name'):
             subnet = self.oneview_client.id_pools_ipv4_subnets.get(self.data.get('subnetUri'))
             for range_uri in subnet['rangeUris']:
-                maybe_resource = self.resource_client.get(range_uri)
-                if maybe_resource['name'] == self.data['name']:
+                maybe_resource = self.resource_client.get_by_uri(range_uri)
+                if maybe_resource.data['name'] == self.data['name']:
                     resource = maybe_resource
-                    break
-
-        self.data['type'] = self.data.get('type', 'Range')
 
         if self.state == 'present':
-            return self.__present(resource)
+            return self._present(resource)
         elif self.state == 'absent':
-            return self.resource_absent(resource)
-
-    def __present(self, resource):
+            if resource:
+                self.current_resource = resource
+            else:
+                self.current_resource = None
+            return self.resource_absent()
+
+    def _present(self, resource):
+        # If no resource was found during get operation, it creates new one
         if not resource:
-            response = self.resource_present(resource, 'id_pools_ipv4_range')
+            response = self.resource_present("id_pools_ipv4_range")
         else:
+            # setting current resource for _update_resource
+            self.current_resource = resource
             # Enabled can be True, False or None. Using not found default 'X' for comparison purposes.
             enabled = self.data.pop('enabled', 'X')
+            # sets update_collector/update_allocator if Given to True.
+            update_collector = self.data.pop('update_collector', 'X')
```

Comment:
Why do you use this 'X' variable..?

Reply:
for default value, its an exception when user do not provide any inputs.

Reply:
updated.